lib/improver/tests/test_nbhood_utilities.py

```diff
@@ -0,0 +1,290 @@
+        data = np.array([[[5., 10., 14., 19., 24.],
+                          [4., 8., 11., 15., 19.],
+                          [3., 6., 8., 11., 14.],
+                          [2., 4., 6., 8., 10.],
+                          [1., 2., 3., 4., 5.]],
+                         [[5., 10., 15., 19., 24.],
+                          [4., 8., 12., 15., 19.],
+                          [3., 6., 9., 11., 14.],
+                          [2., 4., 6., 7., 9.],
+                          [1., 2., 3., 4., 5.]],
+                         [[4., 9., 14., 19., 24.],
+                          [4., 8., 12., 16., 20.],
+                          [3., 6., 9., 12., 15.],
+                          [2., 4., 6., 8., 10.],
+                          [1., 2., 3., 4., 5.]]])
+
+        cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2), (0, 1, 3, 3), (0, 2, 0, 0)),
+            num_time_points=3, num_grid_points=5)
+        result = Utilities.cumulate_array(cube)
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, data)
+
+    def test_for_multiple_realizations_and_times(self):
+        """
+        Test that the vertical and horizontal accumulation produces the
+        intended result when the input cube has multiple times.
+        """
+        data = np.array([[[[5., 10., 14., 19., 24.],
+                           [4., 8., 11., 15., 19.],
+                           [3., 6., 8., 11., 14.],
+                           [2., 4., 6., 8., 10.],
+                           [1., 2., 3., 4., 5.]],
+                          [[4., 9., 14., 19., 24.],
+                           [4., 8., 12., 16., 20.],
+                           [3., 6., 9., 12., 15.],
+                           [2., 4., 6., 8., 10.],
+                           [1., 2., 3., 4., 5.]]],
+                         [[[5., 10., 15., 19., 24.],
+                           [4., 8., 12., 15., 19.],
+                           [3., 6., 9., 11., 14.],
+                           [2., 4., 6., 7., 9.],
+                           [1., 2., 3., 4., 5.]],
+                          [[5., 9., 14., 19., 24.],
+                           [4., 7., 11., 15., 19.],
+                           [3., 5., 8., 11., 14.],
+                           [2., 4., 6., 8., 10.],
+                           [1., 2., 3., 4., 5.]]]])
+
+        cube = set_up_cube(
+            zero_point_indices=(
+                (0, 0, 2, 2), (1, 0, 3, 3), (0, 1, 0, 0), (1, 1, 2, 1)),
+            num_time_points=2, num_grid_points=5, num_realization_points=2)
+
+        result = Utilities.cumulate_array(cube)
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, data)
+
+
+class Test_find_required_lead_times(IrisTest):
+
+    """Test determining of the lead times present within the input cube."""
+
+    def test_basic(self):
+        """Test that a list is returned."""
+        cube = add_forecast_reference_time_and_forecast_period(set_up_cube())
```

Comment:
np.ndarray is returned not a list.

This goes for the following tests as well. It would be better to replace list with array in the description.